tardis/plasma/properties/nlte.py

```diff
@@ -103,3 +105,103 @@ def filter_with_helium_index(atomic_number, ion_number, partition_function,
             columns=['ionization_energy'])
         return partition_function_index, ionization_data_index,\
                partition_function, ionization_data
+
+class HeliumNumericalNLTE(ProcessingPlasmaProperty):
+    outputs = ('helium_population',)
+    '''
+    IMPORTANT: This particular property requires a specific numerical NLTE
+    solver and a specific atomic dataset (neither of which are distributed
+    with Tardis) to work.
+    '''
+    def calculate(self, ion_number_density, electron_densities, t_electrons, w,
+        lines, j_blues, levels, level_boltzmann_factor, t_rad,
+        zeta_data, g_electron, delta, partition_function, ionization_data,
+        beta_rad, g):
+        print 'Performing numerical NLTE calculations.'
+        if len(j_blues)==0:
+            return None
+        heating_rate_data = np.loadtxt(
+            self.plasma_parent.heating_rate_data_file, unpack=True)
+        #Outputting data required by SH module
+        for zone in range(0, len(electron_densities)):
+            output_file = open('He_NLTE_Files/shellconditions_%d.txt' %zone,
+                'w')
+            print>>output_file, ion_number_density.ix[2].sum()[zone]
+            print>>output_file, electron_densities[zone]
+            print>>output_file, t_electrons[zone]
+            print>>output_file, heating_rate_data[zone]
+            print>>output_file, w[zone]
+            print>>output_file, self.plasma_parent.time_explosion
+            print>>output_file, t_rad[zone]
+            print>>output_file, self.plasma_parent.v_inner[zone]
+            print>>output_file, self.plasma_parent.v_outer[zone]
+            output_file.close()
+
+            output_file = open('He_NLTE_Files/abundances_%d.txt' %zone, 'w')
```

Comment:
You should consider using this idiom in the future:

with open('He_NLTE_Files/abundances_%d.txt' %zone, 'w') as fd:
    do stuff with fd

you won't have to remember to close the file then. And it's sort of cleaner. It will also close the file even if an exception is raised inside the body.